git-make-release

```diff
@@ -41,40 +14,44 @@ if [ -n "`git status --porcelain -uno`" ]; then
 fi
 
 # obtain the latest tag and version
-IS_TAGGED=1
-LAST_TAG=`git describe --abbrev=0 --tags` || IS_TAGGED=0
-if [ $IS_TAGGED == 1 ]; then
-  echo "Last tag: ${LAST_TAG}"
-  LAST_VERSION=`echo $LAST_TAG | sed -e 's|^v||'`
-  echo "Last version: ${LAST_VERSION}"
-
-  SHOW_CHANGES_COMMAND="${script_home}/git-show-changes-since-last-tag"
-else
-  LAST_VERSION="initial commit"
-  SHOW_CHANGES_COMMAND="git log --graph"
-  echo "No previous tagged version found."
-fi
-
-# obtain the change log since the latest tag
-echo "Changes since ${LAST_VERSION}:"
-CHANGES=`mktemp`
-echo "============================================================"
-${SHOW_CHANGES_COMMAND} | grep -v "^commit " | grep -v "Author: " | grep -v "Date: " | sed -e 's|^ *||' | tee $CHANGES
-echo "============================================================"
+LAST_TAG=`git tag | tail -n1`
+echo "Last tag: ${LAST_TAG}"
+LAST_VERSION=`echo $LAST_TAG | sed -e 's|^v||'`
+echo "Last version: ${LAST_VERSION}"
 
 # extract new version from CMakeLists.txt
-MAJOR=`grep -i 'set(${PROJECT_NAME}_MAJOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MAJOR_VERSION ||i' -e 's|)$||'`
-MINOR=`grep -i 'set(${PROJECT_NAME}_MINOR_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MINOR_VERSION ||i' -e 's|)$||'`
-PATCH=`grep -i 'set(${PROJECT_NAME}_PATCH_VERSION' ${REPO_TOPLEVEL}/CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_PATCH_VERSION ||i' -e 's|)$||'`
+MAJOR=`grep -i 'set(${PROJECT_NAME}_MAJOR_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MAJOR_VERSION ||i' -e 's|)$||'`
+MINOR=`grep -i 'set(${PROJECT_NAME}_MINOR_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_MINOR_VERSION ||i' -e 's|)$||'`
+PATCH=`grep -i 'set(${PROJECT_NAME}_PATCH_VERSION' CMakeLists.txt | sed -e 's|^set(${PROJECT_NAME}_PATCH_VERSION ||i' -e 's|)$||'`
+
+if [ -z "$MAJOR" -o -z "$MINOR" -o -z "$PATCH" ]; then
```

Comment:
What is the use case of not having a CMakeLists.txt?

Reply:
The use case is having a CMakeLists.txt but not define the version inside as we do normally in our project-template based software. This is done in case of the ldd firmware, which defines its version via `git describe` (like the FPGA firmware does btw).